### ISSUE TITLE
More documentation for site developers needed

### ISSUE THREAD

**Issue by Rask** [HelloZeroNet/ZeroNet#89]
I'm building a ZeroNet blogging platform from scratch using Backbone.js and JSON+Markdown data formats.

I'd like to see more documentation on the ZeroFrame file API and routing, among additions to the code examples provided.

Currently there are copy-pastes CoffeeScript examples available at the documentation page regarding `fileGet` (which I'm using succesfully to read JSON, Markdown and .html template files) and `fileWrite` (which I can't seem to get working properly).

I need to save blog post contents into Markdown files and the post meta (such as date, title, publish status and so on) into a general `posts.json` file. Currently the WebSocket connection is error closed when I'm trying to write a new Markdown file into a site subfolder. Do the files need to exist in the `files` construct of the `content.json` file in order allow `fileWrite` to work properly?

Routing is a bit of a mystery for me in the ZeroFrame API. There is code available in the ZeroBlog source, but I'd like to know how setting the site `iframe`'s URL hash could be propagated to the ZeroFrame and onto the main browser window's URL hash.

Could the documentation be beefed up a bit to include pure JS examples that are not copy-pasted out of context (e.g. containing unknown variables and such)?

All additional information would be awesome. :)

**Comment by HelloZeroNet** [comment]
Yeah, the document definitely needs more work. I have made some modifications in fileWrite recently, its possible as side-effect the file already has to be in content.json before its allows writing (so needs to be created and `zeronet.py siteSign youraddress` executed). I'm going to test it later today. 
Do you have any error in `log/debug.log` file?

**Comment by HelloZeroNet** [comment]
I have tried it and its works for me even for non-exitstent files, updated the sample application with file read/write example:
https://github.com/HelloZeroNet/Documentation/blob/master/example/ZeroFrame/index.html

**Comment by Rask** [comment]
Oh okay. I created the site on another computer and I'm working on it on my home computer. Great to know! I'll test it out.
